tests/Utils/ClearStateTestListener.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace SimpleSAML\Test\Utils;
+
+use PHPUnit_Framework_Test;
+use PHPUnit_Framework_TestSuite;
+
+/**
+ * A PHPUnit test listener that attempts to clear global state and cached SSP configuration between test run
+ */
+class ClearStateTestListener extends \PHPUnit_Framework_BaseTestListener
```

Comment:
Sorry! 😞 

As you can see, my memory doesn't serve me well after a few months... 😓 

I agree with you that the base class that tests can extend is much more flexible, because each test can control what parts of the state it needs clean. Going back to that conversation, I can also see why I preferred transparency, but maybe that benefit comes at a high price. In the end, we should document how to do testing in the project anyway, so maybe transparency isn't that important in this case...

In any case, I don't want you to feel like you have wasted your time because I changed my mind here, so I leave it up to you how to proceed from here. If you don't have time to change the PR to use a base class, we can just merge it now and I'll take responsibility to do the changes myself. If you want to change it before merging the PR, I'll be grateful and owe you a few beers next time we meet each other 😉